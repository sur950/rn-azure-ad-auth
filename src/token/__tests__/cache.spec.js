jest.mock('react-native')
jest.mock('@react-native-async-storage/async-storage')
import TokenCache from '../cache'

const clientId = '123'
const userId = 'userId'

describe('token Cache', () => {
    const cacheInstance = new TokenCache({clientId: clientId, persistent: true})

    it('should be singletone', () => {
        expect((new TokenCache({clientId: '123222'})).clientId).toBe(clientId)
    })

    it('should throw error without client id', () => {
        expect(() => { new TokenCache({wrong: '123222'})} ).toThrow(/Missing/)
    })

    it('should return refresh token', () => {
        expect(cacheInstance.getRefreshToken(userId)).resolves.toMatchSnapshot()
    })

})