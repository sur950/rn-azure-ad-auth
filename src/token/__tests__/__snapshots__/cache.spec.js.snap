// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`token Cache should return refresh token 1`] = `null`;
